File: 2 klasa/python/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potegi.py
#
def silnia_it(n):
    # 0! = 1
    # n! = 1*2 * ... *n
    wynik = 1
    
    for i in range(1, n + 1):
        wynik = wynik * i
        print(wynik)
    return wynik

def main(args):
    n = int(input("Podaj liczbę: "))
    silnia_it(n)
    return 0

if __name__ == '__main__':
    import sys
sys.exit(main(sys.argv))

File: 2 klasa/cpp/silnia.cpp
/*
 * silnia.cpp
 * 
 * Copyright 2018 kl2ag1 <kl2ag1@ubu08>
 */


#include <iostream>
using namespace std;

float silnia_re(int n) {
    if (n == 0) return 1;
    return silnia_re(n-1) * n;
}

int silnia_it(int n){
    int silnia = 1;
    int i;
    for (i = 1; i <= n; i++)
        silnia = silnia * i;
    return silnia;
    }
int main(int argc, char **argv)
{
    int n;
    cout << " Podaj liczbę: "; cin >> n;
    cout << "Wyniki: " << silnia_it(n);
	return 0;
}

